server: add more info to max-go-limit warning

RaduBerinde opened the pull request:
Add the sql memory, cache size, and total memory to the warning issued
when `--max-go-memory` is clamped.

Fixes #130195

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130255)
<!-- Reviewable:end -->

tbg commented on pkg/cli/start.go:
```diff
@@ -744,8 +744,10 @@ func getDefaultGoMemLimit(ctx context.Context) int64 {
 			ctx, severity.WARNING, "recommended default value of "+
 				"--max-go-memory (%s) was truncated to %s, consider reducing "+
-				"--max-sql-memory and / or --cache",
+				"--max-sql-memory (%s) and / or --cache (%s); total system/cgroup memory: %s.",
```
The args are swapped - you want the memory pool size first (since that's the sql mem according to line line 739 above), then `serverCfg.CacheSize` since that's the `--cache`, then `sysMem`.

RaduBerinde replied:
Good catch, thanks!

RaduBerinde committed:
server: add more info to max-go-limit warning

Add the sql memory, cache size, and total memory to the warning issued
when `--max-go-memory` is clamped.

Fixes #130195

RaduBerinde commented:
bors r+

RaduBerinde commented:
@tbg mind stamping with an approval when you get a chance?

tbg reviewed (approved):
bors r+
